Composer/packages/client/src/pages/language-generation/index.tsx

```diff
@@ -35,47 +34,30 @@ interface LGPageProps extends RouteComponentProps<{}> {
 const LGPage: React.FC<LGPageProps> = props => {
   const { state } = useContext(StoreContext);
   const { lgFiles, dialogs } = state;
+
   const path = props.location?.pathname ?? '';
   const { fileId = 'common' } = props;
   const edit = /edit(\/)*$/.test(path);
   const file = lgFiles.find(({ id }) => id === 'common');
+  const navLinks = useMemo(() => {
+    const newDialogLinks = dialogs.map(dialog => {
+      return { id: dialog.id, url: dialog.id, key: dialog.id, name: dialog.displayName };
+    });
+    const mainDialogIndex = newDialogLinks.findIndex(link => link.id === 'Main');
+    const mainDialog = newDialogLinks.find(link => link.id === 'Main');
 
-  const navLinks = useMemo<INavLinkGroup[]>(() => {
-    const subLinks = dialogs.reduce<INavLink>((result, file) => {
-      const item = {
-        id: file.id,
-        key: file.id,
-        name: file.displayName,
-        url: file.id,
-      };
-
-      if (file.isRoot) {
-        result = {
-          ...result,
-          ...item,
-          isExpanded: true,
-        };
-      } else {
-        result.links = result.links || [];
-        result.links.push(item);
-      }
-      return result;
-    }, {} as INavLink);
+    if (mainDialog) {
+      newDialogLinks.splice(mainDialogIndex, 1);
```

Comment:
you can use the return value of `Array.prototype.splice()`, it's exactly the `mainDialog` object you need (wrapped in an array)